test.py

```diff
@@ -22,30 +17,52 @@ def test(folder):
         if entry.is_file() and entry.path.endswith(porth_ext):
             print('[INFO] Testing %s' % entry.path)
 
-            txt_path = entry.path[:-len(porth_ext)] + ".txt"
+            bin_path = entry.path[:-len(porth_ext)] + ".bin"
             expected_output = None
-            with open(txt_path, "rb") as f:
-                expected_output = f.read()
+            with open(bin_path, "rb") as f:
+                bin_file = f.read()
+                index = 0
```

Comment:
The `read` method has an optional `size` in bytes. You can read the chunks of corresponding size without maintaining this `index`.